File: orion-ui/packages/orion-design/src/models/Log.ts
export interface ILog {
  id: string,
  created: Date,
  updated: Date,
  name: string,
  level: number,
  message: string,
  timestamp: Date,
  flowRunId: string,
  taskRunId: string,
}

export class Log implements ILog {
  public readonly id: string
  public readonly created: Date
  public readonly updated: Date
  public name: string
  public level: number
  public message: string
  public timestamp: Date
  public flowRunId: string
  public taskRunId: string

  public constructor(log: ILog) {
    this.id = log.id
    this.created = log.created
    this.updated = log.updated
    this.name = log.name
    this.level = log.level
    this.message = log.message
    this.timestamp = log.timestamp
    this.flowRunId = log.flowRunId
    this.taskRunId = log.taskRunId
  }
}

File: src/prefect/client/schemas/objects/log.py
from typing import (
    Optional,
)
from uuid import UUID

from pydantic import (
    Field,
)
from pydantic_extra_types.pendulum_dt import DateTime

from prefect._internal.schemas.bases import ObjectBaseModel


class Log(ObjectBaseModel):
    """An ORM representation of log data."""

    name: str = Field(default=..., description="The logger name.")
    level: int = Field(default=..., description="The log level.")
    message: str = Field(default=..., description="The log message.")
    timestamp: DateTime = Field(default=..., description="The log timestamp.")
    flow_run_id: Optional[UUID] = Field(
        default=None, description="The flow run ID associated with the log."
    )
    task_run_id: Optional[UUID] = Field(
        default=None, description="The task run ID associated with the log."
    )
